config/config.go

```diff
@@ -140,7 +144,11 @@ func (l *Loader) getResolver() FileResolver {
 // YamlProvider returns function to create Yaml based configuration provider
 func (l *Loader) YamlProvider() ProviderFunc {
 	return func() (Provider, error) {
-		return NewYAMLProviderFromFiles(false, l.getResolver(), l.getFiles()...), nil
+		static := NewYAMLProviderFromFiles(false, l.getResolver(), l.getStaticFiles()...)
+		expanded := NewYAMLProviderWithExpand(false, l.getResolver(), os.LookupEnv, l.getFiles()...)
+
+		// Expanded files will have higher priority than static.
+		return NewProviderGroup("yaml", static, expanded), nil
```

Comment:
Continuing the conversation - this will not work.

Secrets must be able to override keys from other files, that's what they are there for...